### PR TITLE
pagy v9 compatibility, backwards compatible

### PR DESCRIPTION
**Pull request by yshmarov**
Fixes https://github.com/avo-hq/avo/issues/3014
pagy upgrade docs: https://ddnexus.github.io/pagy/changelog/#version-900

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -185,6 +185,6 @@ gem "avo-money_field"
 gem "avo-record_link_field"
 
 # gem "pagy", "< 8.0.0"
-gem "pagy", "> 8"
+gem "pagy", "> 9"
```
Comment (yshmarov) [inline Gemfile:188]: We don't have to commit the Gemfile changes right now, to ensure that we don't break other apps.

**Comment by adrianthedev** [comment]
Thank you @yshmarov!
We appreciate your contribution ❤️

File: Gemfile (hunk 1 above)
Comment (Paul-Bob) [reply Gemfile]: It's ok, this Gemfiles does not reflect on avo's dependencies, those are specified in `avo.gemspec`

**Review by Paul-Bob — APPROVED** [review]
Thank you @yshmarov!

File: config/initializers/pagy.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 require "pagy/extras/trim"
 require "pagy/extras/countless"
+require "pagy/extras/size"
```
Comment (Paul-Bob) [inline config/initializers/pagy.rb:3]: Let's keep this for now.

I'm creating an issue to use integer sizes in favor of using the array legacy API.

**Comment by iainbeeston** [comment]
This has actually broken my rails application that uses avo. On app startup I get a `cannot load such file -- pagy/extras/size`
error from line 3 of `config/initializers/pagy.rb`in avo. Looking at pagy on GitHub, it looks like `pagy/extras/size` was only added a week ago in pagy 9.0 and older versions of page don't have this file (and will crash). I'm using 7.0.11.

I can see #3049 there's a change scheduled that removes the need for this PR, can that get prioritised or this PR reverted please?

**Comment by adrianthedev** [comment]
yup. On it @iainbeeston

**Comment by Paul-Bob** [comment]
Fixed on 3.10.9

**Comment by simi** [comment]
Any chance to backport to 2.x?

**Pull request merged**